flake.nix

```diff
@@ -0,0 +1,69 @@
+{
+  description = "Apple M1 support for NixOS";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-parts.url = "github:hercules-ci/flake-parts";
```

Comment:
I would prefer a vanilla flake. Do we need the extra abstraction and dependency?

Reply:
This can make (especially) larger projects cleaner and easier to manage, but since our `flake.nix` is simple, a vanilla flake should be good enough.